### DISCUSSION TITLE
LoadListDataAsStreamAsync is unable to read the items inside a folder inside my sharepoint custom list

### DISCUSSION THREAD

**Discussion by mvcsharepointdev** [pnp/pnpcore#1403] [Q&A] [answered]
I have the following:-

1) SharePoint custom list which contain 1559 items.
2) and it contain a folder named "Closed" which contain 181 items

now i am using this code to get all the 1559, but this will only return 1378 items,,, so seems it skipped the items inside the "Closed' folder" :- 

            var workOrderList = context.Web.Lists.GetByTitle("Work Orders", p => p.Title,
                                                                 p => p.Fields.QueryProperties(p => p.InternalName,
                                                                                               p => p.FieldTypeKind,
                                                                                               p => p.TypeAsString,
                                                                                               p => p.Title));

            // Build a query that only returns the Title field for the first 20 items where the Title field starts with "Item1"
            string viewXml = @"<View>
                    <ViewFields>
                      <FieldRef Name='Title' />
                      <FieldRef Name='FileLeafRef' />
                    </ViewFields>
                    <Query>
                    </Query>
                   </View>";

            // Load all the needed data using paged requests
            bool paging = true;
            string nextPage = null;
            while (paging)
            {
                var output = await workOrderList.LoadListDataAsStreamAsync(new RenderListDataOptions()
                {
                    ViewXml = viewXml,
                    RenderOptions = RenderListDataOptionsFlags.ListData,
                    Paging = nextPage ?? null,
                }).ConfigureAwait(false);

                if (output.ContainsKey("NextHref"))
                {
                    nextPage = output["NextHref"].ToString().Substring(1);
                }
                else
                {
                    paging = false;
                }
            }

            // Iterate over the retrieved list items
            foreach (var listItem in workOrderList.Items.AsRequested())
            {
                // Do something with the list item
                i++;
            }

any advice, how i cam get all the list items inside the root folder and inside all sub folders?

**Comment by jansenbe** [comment, marked as answer]
@mvcsharepointdev : update the view element in your CAML query to be like this: `<View Scope='Recursive'>`

**Reply by mvcsharepointdev** [reply]
@jansenbe ok worked thanks
